cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup.go

```diff
@@ -111,17 +127,27 @@ func (ng *nodegroup) DeleteNodes(nodes []*apiv1.Node) error {
 		}
 
 		machine.Annotations[machineDeleteAnnotationKey] = time.Now().String()
-
 		if _, err := ng.machineapiClient.Machines(machine.Namespace).Update(machine); err != nil {
-			return fmt.Errorf("failed to update machine %s/%s: %v", machine.Namespace, machine.Name, err)
+			return err
+		}
+
+		if err := ng.scalableResource.SetSize(int32(replicas - 1)); err != nil {
+			delete(machine.Annotations, machineDeleteAnnotationKey)
+			// Log errors as warnings from Update()
+			// because no action is taken even if the
+			// annotation persists until the replica count
+			// is modified during a deletion.
+			_, updateErr := ng.machineapiClient.Machines(machine.Namespace).Update(machine)
```

Comment:
This might actually fail since after calling `Update(machine)` on line 115 the `machine` object will get outdated. You might as well do
```go
newMachine, err := ng.machineapiClient.Machines(machine.Namespace).Update(machine)
if err != nil {
...
}
machine = newMachine
```

Reply:
This is end of the loop. We don't do anything more with `machine`.

Reply:
OK, we were looking at different lines. I believe your comment was against line 130 which as discussed on Slack we've since resolved in https://github.com/openshift/kubernetes-autoscaler/commit/35cc7965d8785797674017a8d08f3b3ccca27706